src/lib/cbe/cbe-primitive.ads

```diff
@@ -0,0 +1,94 @@
+--
+-- Copyright (C) 2019 Genode Labs GmbH, Componolit GmbH, secunet AG
+--
+-- This file is part of the Consistent Block Encrypter project, which is
+-- distributed under the terms of the GNU Affero General Public License
+-- version 3.
+--
+
+pragma Ada_2012;
+
+with CBE.Request;
+
+package CBE.Primitive
+with Spark_Mode
+is
+	pragma Pure;
+
+	use type Request.Operation_Type;
+	use type Request.Success_Type;
+	use type Request.Block_Number_Type;
+
+	type Index_Type  is mod 2**64;
```

Comment:
This declaration hides `CBE.Index_Type`. You might either rename it or even drop this declaration completely.